src/mame/sinclair/specnext.cpp

```diff
@@ -807,11 +835,87 @@ void specnext_state::memory_change(u16 port, u8 data) // port_memory_change_dly
 	m_port_1ffd_special_old = port_1ffd_special();
 }
 
+void specnext_state::update_video_mode()
+{
+	const u16 MUST_BE_447 = 455; // FIXME reconfigure screen width to lower value than present makes bgfx crash
+	std::string machine_name;
+	if (BIT(m_nr_03_machine_timing, 2))
+	{
+		machine_name = "Pentagon";
+		m_video_timings = // Pentagon is always 50 Hz
+		{
+			0, 448 + 3 - 12, 16, 47, 63, 128, MUST_BE_447,
+			0, 319, 1, 14, 15, 80, 319,
+			76, 435, 24, 311, 24 + 0
+		};
+		m_nr_05_5060 = 0;
+	}
+	else if (BIT(m_nr_03_machine_timing, 1))
+	{
+		machine_name = BIT(m_nr_03_machine_timing, 0) ? "+3" : "128K";
+		if (!m_nr_05_5060)
+			m_video_timings = // 128K 50 Hz
+			{
+				0, u16(!BIT(m_nr_03_machine_timing, 0) ? 136 + 4 - 12 /* 128 */ : 136 + 2 - 12 /* +3 */), 16, 47, 95, 136, 455,
+				0, 1, 1, 4, 7, 64, 310,
+				88, 447, 16, 303, 16 + 4
+			};
+		else
+			m_video_timings = // 128K 60 Hz
+			{
+				0, u16(!BIT(m_nr_03_machine_timing, 0) ? 136 + 4 - 12 /* 128 */ : 136 + 2 - 12 /* +3 */), 16, 47, 95, 136, 455,
+				0, 0, 1, 4, 7, 40, 263,
+				88, 447, 16, 255, 16 + 0
+			};
+	}
+	else
+	{
+		machine_name = "48K";
+		if (!m_nr_05_5060)
+			m_video_timings = // 48K 50 Hz
+			{
+				0, 128 + 0 - 12, 16, 47, 95, 128, MUST_BE_447,
+				0, 0, 1, 4, 7, 64, 311,
+				80, 439, 16, 303, 16 + 4
+			};
+		else
+			m_video_timings = // 48K 60 Hz
+			{
+				0, 128 + 0 - 12, 16, 47, 95, 128, 447,
+				0, 0, 1, 4, 7, 40, 263,
+				80, 439, 16, 255, 16 + 0
+			};
+	}
+
+	const bool is_hdmi = ~m_io_video.read_safe(1) & 1;
+	const int left = m_video_timings.min_hactive << 1;
+	const int top = m_video_timings.min_vactive;
+	const int width = (m_video_timings.max_hc + 1) << 1;
+	const int height = m_video_timings.max_vc + 1;
+	m_clip256x192 = rectangle(left, left + (256 << 1), top, top + 192);
+	m_clip320x256 = rectangle(left - (32 << 1), left + ((256 + 32) << 1), top - 32, top + 192 + 32);
+
+	m_screen->configure(width, height
+		, is_hdmi
+			? rectangle(m_video_timings.hdmi_xmin << 1, m_video_timings.hdmi_xmax << 1,m_video_timings.hdmi_ymin, m_video_timings.hdmi_ymax)
+			: m_clip320x256
+		, HZ_TO_ATTOSECONDS(28_MHz_XTAL / 2) * width * height);
+	m_ula->set_raster_offset(left, top);
+	m_lores->set_raster_offset(left, top);
+	m_tiles->set_raster_offset(left, top);
+	m_layer2->set_raster_offset(left, top);
+	m_sprites->set_raster_offset(left, top);
+
+	m_eff_nr_03_machine_timing = m_nr_03_machine_timing;
+	m_eff_nr_05_5060 = m_nr_05_5060;
+	popmessage("%s: %s %dHz", machine_name, is_hdmi ? "HDMI" : "VGA", m_nr_05_5060 ? 60 : 50);
```

Comment:
This should be LOGMASKED() or similar rather than just shown to the user.

Reply:
IMHO this way it gives good user experience. Similar how your TV set show such message.

Reply:
If we want to pursue this road then `popmessage` isn't the right tool positioning-wise. 
Should be a specific user opt-in function, background color should be more transparent, and be between the [action-safe and title-safe area](https://en.wikipedia.org/wiki/Safe_area_(television)). 
Consider also that `popmessage` has this tendency of sticking right inside in the title-safe area for too much (specifically for `temp.length() / 40 + 2`, in seconds), and this isn't a message that should stick for this long ...

Reply:
I've switched to general LOG